File: TME5/Imagier/Imagier/ViewController.m
//
//  ViewController.m
//  Imagier
//
//  Created by m2sar on 28/10/2016.
//  Copyright © 2016 UPMC. All rights reserved.
//

#import "ViewController.h"
#import "MaVue.h"

@interface ViewController ()

@end

@implementation ViewController

MaVue *v;

- (void)viewDidLoad {
    [super viewDidLoad];
    // Do any additional setup after loading the view, typically from a nib.
    v = [[MaVue alloc] initWithFrame:[[UIScreen mainScreen] bounds]];
    [v setBackgroundColor:[UIColor whiteColor]];
    [self setView:v];
    
    [v.sliderScroll addTarget:self action:@selector(sliderAction:) forControlEvents:UIControlEventValueChanged];
    
    [v.next setAction:@selector(navigationAction:)];
    [v.previous setAction:@selector(navigationAction:)];
    
    v.i = 0;
    
    [v.sliderScroll setValue:0.2];
    [v.scroll setZoomScale:0.0 animated:YES];
    [v.percent setText:[NSString stringWithFormat:@"%0.f%%", v.sliderScroll.value*100]];
    [v.imageName setText:[NSString stringWithFormat:@"%@",v.imageArray[0]]];
    [v.toolBar setBarTintColor:[UIColor whiteColor]];
    
    [v release];
}


- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}

- (void) sliderAction: (UISlider *)sender{
    [v.scroll setZoomScale:sender.value];
    [v.percent setText:[NSString stringWithFormat:@"%0.f%%", v.sliderScroll.value*100]];
}

- (IBAction) navigationAction: (UIBarButtonItem *)sender {
    if(sender == v.previous){
        if(v.i > 0){
            v.i--;
        }else{
            v.i = v.maxArray - 1;
        }
    }else{
        if(v.i < v.maxArray - 1){
            v.i++;
        }else{
            v.i = 0;
        }
    }
    
    [v.sliderScroll setValue:v.imagePercent[v.i].floatValue];
    [v.percent setText:[NSString stringWithFormat:@"%0.f%%", v.sliderScroll.value*100]];
    
    [v.imageView removeFromSuperview];
    v.imageView = [[UIImageView alloc] initWithImage:[UIImage imageNamed:v.imageArray[v.i]]];
    
    [v.scroll setZoomScale:v.imagePercent[v.i].floatValue];
    
    [v.scroll addSubview:v.imageView];
    [v.imageName setText:v.imageArray[v.i]];
    
}

/*
- (void) viewWillTransitionToSize:(CGSize)size withTransitionCoordinator:(id<UIViewControllerTransitionCoordinator>)coordinator{
    [v drawInFormat:size];
}*/

@end

File: TME6/Asteroide/Asteroide/Asteroide/ViewController.swift
//
//  ViewController.swift
//  Asteroide
//
//  Created by m2sar on 07/11/2016.
//  Copyright © 2016 UPMC. All rights reserved.
//

import UIKit

class ViewController: UIViewController {
    let vGame = GameView(frame: UIScreen.main.bounds)
    let vScore = ScoreView(frame: UIScreen.main.bounds)
    let vPref = PrefView(frame: UIScreen.main.bounds)
    
    override func viewDidLoad() {
        super.viewDidLoad()
        // Do any additional setup after loading the view, typically from a nib.
        
        
        let backgroundImage = UIImageView(frame: UIScreen.main.bounds)
        backgroundImage.image = UIImage(named: "fond-asteroides")
        
        self.view.addSubview(backgroundImage)//0
        self.view.addSubview(vGame)//1
        self.view.addSubview(vScore)//2
        self.view.addSubview(vPref)//3
        
        vScore.isHidden = true
        vPref.isHidden = true
        
        vPref.levelSelected = 1
        
        self.vGame.playButton.addTarget(self.vGame, action: #selector(vGame.play), for: .touchDown)
        self.vGame.scoreButton.addTarget(self.vGame, action: #selector(vGame.scores), for: .touchDown)
        self.vGame.prefButton.addTarget(self.vGame, action: #selector(vGame.preference), for: .touchDown)
        
        
    }

    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }

}

